### DISCUSSION TITLE
Trigger function in dropdown

### DISCUSSION THREAD

**Discussion by xypnox** [floating-ui/floating-ui#1877] [Q&A] [answered]
The dropdown menu example in https://codesandbox.io/s/admiring-lamport-5wt3yg?file=/src/DropdownMenu.tsx is pretty complete in the sense that it supports almost all user interactions including the keyboard, however, I can't figure how to attach the proper functions to trigger when an item is selected from the menu.

Adding onClick to the menu item doesn't work (throws error of it not being in the type of props), and it won't work with keyboard.

How can I add a specific trigger function with the items?

**Comment by atomiks** [comment, marked as answer]
Line 232 has the `onClick` handler in `getItemProps()`, which is where you can add your logic

**Reply by atomiks** [reply]
CodeSandbox example: https://codesandbox.io/s/xenodochial-taussig-ycz7qg?file=/src/DropdownMenu.tsx
